--- origsrc/librep-0.91.0/Makedefs.in	2010-09-24 14:19:26.000000000 -0500
+++ src/librep-0.91.0/Makedefs.in	2011-02-06 23:18:57.435036300 -0600
@@ -24,6 +24,7 @@ SHELL=/bin/sh
 
 top_srcdir=@top_srcdir@
 srcdir=@srcdir@
+top_builddir=@top_builddir@
 
 prefix=@prefix@
 exec_prefix=@exec_prefix@
@@ -79,11 +80,11 @@ HAVE_UNIX=@HAVE_UNIX@
 .PHONY: clean realclean install
 
 rep_prog = $(top_builddir)/src/rep
-COMPILE_ENV = REPLISPDIR=$(top_builddir)/lisp \
+COMPILE_ENV = REPLISPDIR=$(top_srcdir)/lisp \
 	      REP_DL_LOAD_PATH=$(top_builddir)/src/.libexec \
 	      REPDOCFILE=$(top_builddir)/doc-strings
 
-include $(top_srcdir)/rules.mk
+include $(top_builddir)/rules.mk
 rep_LIBTOOL=@LIBTOOL@
 
 # Rule for ``normal'' C objects
--- origsrc/librep-0.91.0/Makefile.in	2010-09-24 14:19:26.000000000 -0500
+++ src/librep-0.91.0/Makefile.in	2011-02-06 23:12:12.752038400 -0600
@@ -49,13 +49,13 @@ install : all installdirs
 	  ( cd $$dir && $(MAKE) $@ ) || exit 1; \
 	done
 	$(INSTALL_DATA) doc-strings $(DESTDIR)$(repexecdir)
-	$(INSTALL_SCRIPT) emulate-gnu-tar $(DESTDIR)$(repexecdir)
+	$(INSTALL_SCRIPT) $(top_srcdir)/emulate-gnu-tar $(DESTDIR)$(repexecdir)
 	$(INSTALL_SCRIPT) libtool $(DESTDIR)$(repcommonexecdir)
 	$(INSTALL_DATA) rules.mk $(DESTDIR)$(repcommonexecdir)
-	$(INSTALL_SCRIPT) install-aliases $(DESTDIR)$(repcommonexecdir)
-	$(INSTALL_DATA) rep-debugger.el $(DESTDIR)$(emacssitelispdir)
+	$(INSTALL_SCRIPT) $(top_srcdir)/install-aliases $(DESTDIR)$(repcommonexecdir)
+	$(INSTALL_DATA) $(top_srcdir)/rep-debugger.el $(DESTDIR)$(emacssitelispdir)
 	mkdir -p $(DESTDIR)$(libdir)/pkgconfig
-	$(INSTALL_DATA) $(top_srcdir)/librep.pc $(DESTDIR)$(libdir)/pkgconfig/
+	$(INSTALL_DATA) librep.pc $(DESTDIR)$(libdir)/pkgconfig/
 
 installdirs : mkinstalldirs
 	$(SHELL) $< $(DESTDIR)$(repdir)/$(version) \
--- origsrc/librep-0.91.0/lisp/Makefile.in	2010-09-24 14:19:26.000000000 -0500
+++ src/librep-0.91.0/lisp/Makefile.in	2011-02-06 23:12:36.191014000 -0600
@@ -47,10 +47,10 @@ check : all
 install : all installdirs
 	for d in $(INSTALL_DIRS); do \
 	  for f in  $(foreach x,$(INSTALL_FILES),$$d/$(x)); do \
-	    $(INSTALL_DATA) $$f $(DESTDIR)$(replispdir)/$$d; \
+	    $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)$(replispdir)/$$d; \
 	  done; \
 	done
-	$(SHELL) $(top_srcdir)/install-aliases -l . $(DESTDIR)$(replispdir)
+	$(SHELL) $(top_srcdir)/install-aliases -l $(srcdir) $(DESTDIR)$(replispdir)
 
 installdirs : mkinstalldirs
 	$(SHELL) $< $(foreach x,$(INSTALL_DIRS),$(DESTDIR)$(replispdir)/$(x))
--- origsrc/librep-0.91.0/man/Makefile.in	2010-09-24 14:19:26.000000000 -0500
+++ src/librep-0.91.0/man/Makefile.in	2011-02-06 23:33:13.245324600 -0600
@@ -49,7 +49,7 @@ install : librep.info installdirs
 	  $(INSTALL_DATA) $$f $(DESTDIR)$(infodir); \
 	done
 	-install-info --info-dir=$(DESTDIR)$(infodir) librep.info
-	for f in rep*.1; do \
+	for f in $(srcdir)/rep*.1; do \
 		gzip $$f -c > $$f.gz; \
 		$(INSTALL_DATA) $$f.gz $(DESTDIR)$(prefix)/share/man/man1/; \
 	done
--- origsrc/librep-0.91.0/src/Makefile.in	2010-09-24 14:19:26.000000000 -0500
+++ src/librep-0.91.0/src/Makefile.in	2011-02-06 23:28:03.779274000 -0600
@@ -27,7 +27,7 @@ COMMON_SRCS =	continuations.c datums.c d
 		streams.c structures.c symbols.c tuples.c values.c weak-refs.c
 UNIX_SRCS =	unix_dl.c unix_files.c unix_main.c unix_processes.c
 
-INSTALL_HDRS = rep.h rep_lisp.h rep_regexp.h rep_subrs.h rep_gh.h rep_config.h
+INSTALL_HDRS = rep.h rep_lisp.h rep_regexp.h rep_subrs.h rep_gh.h
 
 SRCS = $(COMMON_SRCS) $(UNIX_SRCS)
 OBJS = $(SRCS:.c=.lo)
@@ -103,6 +103,7 @@ install : all installdirs
 	for i in $(INSTALL_HDRS); do \
 	  $(INSTALL_DATA) $(srcdir)/$$i $(DESTDIR)$(includedir)/rep/; \
 	done
+	$(INSTALL_DATA) rep_config.h $(DESTDIR)$(includedir)/rep/; \
 	$(SHELL) $(top_srcdir)/install-aliases -c . $(DESTDIR)$(repexecdir)
 
 installdirs : mkinstalldirs
--- origsrc/librep-0.91.0/src/fake-libexec	2010-09-24 14:19:26.000000000 -0500
+++ src/librep-0.91.0/src/fake-libexec	2011-02-06 23:13:16.918898100 -0600
@@ -4,6 +4,7 @@
 
 libdir="`pwd`/.libs"
 libexecdir="./.libexec"
+srcdir="`dirname $0`"
 
 libs="rep.io.db.gdbm rep.io.db.sdbm rep.i18n.gettext rep.io.readline \
       rep.lang.record-profile rep.data.tables rep.io.timers \
@@ -18,7 +19,7 @@ for f in $libs; do
   for b in `( cd $libdir && echo *\`basename $g\`* )`; do
     rm -f $dest/$b
     echo "$src/$b -> $dest/$b"
-    ../mkinstalldirs $dest
+    $srcdir/../mkinstalldirs $dest
     ln -s $src/$b $dest/$b
   done
 done
